app/Models/User.php

```diff
@@ -326,6 +327,23 @@ public function getActivitylogOptions(): LogOptions
             ->dontSubmitEmptyLogs();
     }
 
+    // roles
+
+    /**
+     * @return DbCollection<int, Role>
+     */
+    public function getVisibleRoles(): Collection
+    {
+        /** @var DbCollection|Role[] $roles */
+        $roles = $this->roles()
+            ->where('display', '>', 0)
+            ->orderBy('display')
+            ->orderBy('name')
```

Comment:
I wonder if there's a better solution to this. `Developer` shares the same display order as `Artist` and `Writer`, but an Artist/Developer would always show Artist, and a Writer/Developer would always show Developer.

I'm also uncertain how the promotion from `Junior Developer` to `Developer` to `Veteran Developer` works. If the old role is not removed, `Developer` would display instead of `Veteran Developer`.

Reply:
The best solution is probably more granular display orders. I think we probably have a rough mental model for how roles should work from a hierarchy point of view.

In the examples given, I'd argue for both Artist/Developer and Writer/Developer, the Artist or Writer roles should be removed and the Developer role should remain, as those two roles will have a subset of the Developer role's abilities.

I'm also unsure how the promotion from Junior Developer to Developer to Veteran Developer works. It seems like we need some kind of "promotion" control in Filament that can intelligently handle this for users responsible for doing promotions. For example, if a Writer is promoted to Developer, the Writer role can be removed.